arch/x86/core/ia32/swap.S

```diff
@@ -303,6 +303,20 @@ CROHandlingDone:
 
 	movl    %eax, _kernel_offset_to_current(%edi)
 
+#if defined(CONFIG_X86_USE_THREAD_LOCAL_STORAGE)
+	pushl	%eax
+
+	/* Push EAX (thread pointer) as argument to z_x86_tls_update_gdt() */
+	pushl	%eax
+	call	z_x86_tls_update_gdt
+
+	/* Since segment descriptor has changed, need to reload */
+	movw	$GS_TLS_SEG, %ax
+	movw	%ax, %gs
```

Comment:
Why not use the same technique as on x86-64 here? I think you can set the gs base address using the same MSR technique and avoid messing with the GDT.

Reply:
All the documents I could find so far says that FS_BASE/GS_BASE MSRs are x86-64 only. And that `wrfsbase` and `wrgsbase` are also x86-64 only.

Reply:
Yeah, they aren't available on classic 32-bit processors (which have no MSRs at all, I've been told). They are available in 32-bit mode on many processors, but having a single version that runs on all 32-bit processors seems best. If multi-core support is ever desired, a GDT-entry per-core could be used.

Reply:
our 32-bit x86 port is for true 32-bit x86 processors like Lakemont.
32-bit mode on a 64-bit processor isn't something we care much about, just use the 64-bit port.
We have no plans for SMP on 32-bit x86 at this time.